src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -90,6 +131,76 @@ public bool TryReadToInternal(out ReadOnlySequence<T> sequence, T delimiter, boo
             return false;
         }
 
+        /// <summary>
+        /// Try to read everything up to the given <paramref name="delimiter"/>, ignoring delimiters that are
+        /// preceeded by <paramref name="skipDelimiter"/>.
+        /// </summary>
+        /// <param name="sequence">The read data, if any.</param>
+        /// <param name="delimiter">The delimiter to look for.</param>
+        /// <param name="skipDelimiter">If found prior to <paramref name="delimiter"/> it will skip that occurence.</param>
+        /// <param name="advancePastDelimiter">True to move past the <paramref name="delimiter"/> if found.</param>
+        /// <returns>True if the <paramref name="delimiter"/> was found.</returns>
+        public bool TryReadTo(out ReadOnlySequence<T> sequence, T delimiter, T skipDelimiter, bool advancePastDelimiter = true)
+        {
+            return TryReadToInternal(out sequence, delimiter, skipDelimiter, advancePastDelimiter);
+        }
+
+        public bool TryReadToInternal(out ReadOnlySequence<T> sequence, T delimiter, T skipDelimiter, bool advancePastDelimiter, int skip = 0)
+        {
+            BufferReader<T> copy = this;
+            if (skip > 0)
+            {
+                if (skip > 1)
+                {
+                    Advance(skip - 1);
+                }
+
+                TryRead(out T prior);
+                if (prior.Equals(skipDelimiter) && !End)
+                {
+                    Advance(1);
+                }
+            }
+
+            ReadOnlySpan<T> remaining = UnreadSpan;
+            bool shouldSkipDelimiter = false;
+
+            while (!End)
+            {
+                int index = remaining.IndexOf(delimiter);
+                if (index != -1)
+                {
+                    if ((shouldSkipDelimiter && index == 0) || (index > 0 && remaining[index - 1].Equals(skipDelimiter)))
```

Comment:
If I am understanding this correctly, there could be an issue here:

Suppose this is your input: `abc\"de"`

The initial search finds index = 4. and we have the skipDelimiter (`\`) at index 3. So we call `TryReadToSlow`.

We then advance by 3 (`skip - 1`), check if we reached skipDelimiter, and advance by 1, if we have. Now, remaining contains: `"de"`
Searching for quotation within the loop:
`shouldSkipDelimiter = false`
`index = 0`

And now we return true saying we found the string, `abc\`, which is incorrect.

Reply:
I'll make sure to test this case.

Reply:
Stupid mistake on my part, I didn't actually copy the "move it forward" part.